react/features/base/sounds/middleware.any.ts

```diff
@@ -71,3 +77,53 @@ function _stopSound({ getState }: IStore, soundId: string) {
         logger.warn(`STOP_SOUND: no sound found for id: ${soundId}`);
     }
 }
+
+/**
+ * Returns whether the language is supported for audio messages.
+ *
+ * @param {string} language - The requested language.
+ * @returns {boolean}
+ */
+function isLanguageSupported(language: string): Boolean {
+    return Boolean(AudioSupportedLanguage[language as keyof typeof AudioSupportedLanguage]);
+}
+
+/**
+ * Checking if it's necessary to reload the translated files.
+ *
+ * @param {string} language - The next language.
+ * @param {string} prevLanguage - The previous language.
+ * @returns {boolean}
+ */
+function shouldReloadAudioFiles(language: string, prevLanguage: string): Boolean {
+    const isNextLanguageSupported = isLanguageSupported(language);
+    const isPrevLanguageSupported = isLanguageSupported(prevLanguage);
+
+    return (
+
+        // From an unsupported language (which defaulted to English) to a supported language (that isn't English).
+        isNextLanguageSupported && language !== AudioSupportedLanguage.en && !isPrevLanguageSupported
+    ) || (
+
+        // From a supported language (that wasn't English) to English.
+        !isNextLanguageSupported && isPrevLanguageSupported && prevLanguage !== AudioSupportedLanguage.en
+    ) || (
+
+        // From a supported language to another.
+        isNextLanguageSupported && isPrevLanguageSupported
+    );
+}
+
+/**
+ * Set up state change listener for language.
+ */
+StateListenerRegistry.register(
+    () => i18next.language,
+    (language, { dispatch }, prevLanguage): void => {
+
+        if (language !== prevLanguage && shouldReloadAudioFiles(language, prevLanguage)) {
+            fetchE2eeAudioFiles(dispatch);
```

Comment:
Shouldn't we have a single point where all sound files are gathered? Feels weird to add categories like this.

Reply:
As it stands right now, each reducer registers its necessary audio files on APP_WILL_MOUNT and unregisters them on APP_WILL_UNMOUNT, so I extracted those dispatches in these functions that can be reused in this language state listener. So far these are the only audio files in need of localization. Would you like for me to refactor that?